language/testing-infra/e2e-tests/src/common_transactions.rs

```diff
@@ -72,6 +72,126 @@ pub static EMPTY_SCRIPT: Lazy<Vec<u8>> = Lazy::new(|| {
         .expect("Failed to compile")
 });
 
+pub static MULTI_AGENT_SWAP_SCRIPT: Lazy<Vec<u8>> = Lazy::new(|| {
+    let code = "
+    import 0x1.DiemAccount;
+    import 0x1.Signer;
+    import 0x1.XDX;
+    import 0x1.XUS;
+
+    // Alice and Bob agree on the value of amount_xus and amount_xdx off-chain.
+    main(alice: signer, bob: signer, amount_xus: u64, amount_xdx: u64) {
+        // First, Alice pays Bob in currency XUS.
+        let alice_withdrawal_cap: DiemAccount.WithdrawCapability;
+        let bob_withdrawal_cap: DiemAccount.WithdrawCapability;
+        let alice_addr: address;
+        let bob_addr: address;
+
+        alice_withdrawal_cap = DiemAccount.extract_withdraw_capability(&alice);
+        bob_addr = Signer.address_of(&bob);
+        DiemAccount.pay_from<XUS.XUS>(
+            &alice_withdrawal_cap, move(bob_addr), move(amount_xus), h\"\", h\"\"
+        );
+        DiemAccount.restore_withdraw_capability(move(alice_withdrawal_cap));
+
+        // Then, Bob pays Alice in currency XDX.
+        bob_withdrawal_cap = DiemAccount.extract_withdraw_capability(&bob);
+        alice_addr = Signer.address_of(&alice);
+        DiemAccount.pay_from<XDX.XDX>(
+            &bob_withdrawal_cap, move(alice_addr), move(amount_xdx), h\"\", h\"\"
+        );
+        DiemAccount.restore_withdraw_capability(move(bob_withdrawal_cap));
+        return;
+    }
+";
+
+    let compiler = Compiler {
+        address: account_config::CORE_CODE_ADDRESS,
+        skip_stdlib_deps: false,
+        extra_deps: vec![],
+    };
+    compiler
+        .into_script_blob("file_name", code)
+        .expect("Failed to compile")
+});
+
+pub static MULTI_AGENT_P2P_SCRIPT: Lazy<Vec<u8>> = Lazy::new(|| {
```

Comment:
is it required to have all signed defined in the script for using multi-agent transaction?
for example, could we use multi-agent txn with one secondary signer for existing peer_to_peer script?

Reply:
@xli Yes it is required that all signers are defined in the argument list of the script function for using multi-agent txns. This is because the validator code creates a signer type argument for each signer, and append this list of signer type arguments to the beginning of argument list. That means multi-agent txns have to use new scripts with multiple signers, not existing peer_to_peer script.